### PR TITLE
Added intranet_uplink_mtu field in vmc_sddc resource. 

### LINKED ISSUE #68: Feature Request : Configure SDDC Intranet Uplink MTU setting via terraform
Could support please be added to the provider to set the Intranet Uplink MTU for the SDDC (SDDC > Networking & Security > Global Configuration > Intranet Uplink MTU). 

With the MTU set to the default 1500 the throughput becomes constrained between the AWS Connected VPC and services hosted on the on-premises site (via DX); we have seen significant throughput increases using Jumbo Frames (e.g. backup streams to EC2 instances) however at the moment we have to manually set this value. It would be useful for us to be able to manage this value using the terraform provider.

### PR DESCRIPTION
**Pull request by saloneerege**
1. intranet_uplink_mtu field is stored in the terraform state file during SDDC creation with the default value of 1500 and then can be updated by the user. 
2. Moved getNSXTReverseProxyURLConnector method to utils.go for vmc_sddc and vmc_public_ip resource. 
3. Updated the public IP resource to use the new Client NewDefaultCloudServiceVmcOnAwsPublicIpClient as a part of the update in nsxt-vmc-aws-integration to v0.4.1

### CODE REVIEW

File: vmc/utils.go [hunk 1]
```diff
@@ -71,3 +74,17 @@ func expandMsftLicenseConfig(config []interface{}) *model.MsftLicensingConfig {
 	licenseConfig = model.MsftLicensingConfig{MssqlLicensing: &mssqlLicensing, WindowsLicensing: &windowsLicensing}
 	return &licenseConfig
 }
+
+func getNSXTReverseProxyURLConnector(nsxtReverseProxyUrl string) (client.Connector, error) {
+	apiToken := os.Getenv(APIToken)
```
Comment (rchadalawada) [inline vmc/utils.go:79]: What if env is not set?

File: vmc/resource_vmc_sddc.go [hunk 2]
```diff
@@ -13,6 +13,8 @@ import (
 	"github.com/hashicorp/terraform/helper/schema"
 	"github.com/hashicorp/terraform/helper/validation"
 	"github.com/vmware/vsphere-automation-sdk-go/lib/vapi/std/errors"
+	nsxtapi "github.com/vmware/vsphere-automation-sdk-go/services/nsxt-vmc-aws-integration/api"
```
Comment (shylasrinivas) [inline vmc/resource_vmc_sddc.go]: probably better to rename this to nsxtintegrationapi or nsxtintapi just to distinguish from nsxt apis

File: vmc/utils.go (hunk 1 above)
Comment (shylasrinivas) [reply vmc/utils.go:79]: Token is validated and set into env variable when terraform is first initialized - https://github.com/vmware/terraform-provider-vmc/blob/master/vmc/provider.go#L81. Terraform apply will error out if it's not set.

**Review by shylasrinivas — APPROVED** [review]

**Review by rchadalawada — APPROVED** [review]

File: vmc/resource_vmc_sddc.go (hunk 2 above)
Comment (saloneerege) [reply vmc/resource_vmc_sddc.go]: Sure. I'll update it

Comment (saloneerege) [reply vmc/resource_vmc_sddc.go]: Done

**Review by shylasrinivas — APPROVED** [review]

**Pull request merged**